src/views/manage-identities/ManageIdentities.tsx

```diff
@@ -0,0 +1,403 @@
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Flex,
+  Text,
+  Input,
+  Alert,
+  AlertIcon,
+  AlertTitle,
+  AlertDescription,
+} from "@chakra-ui/react";
+import { useContext, useState, useEffect } from "react";
+import { useMutation } from "@tanstack/react-query";
+import { ethers } from "ethers";
+import { EvmAddress, TokenNameItem, UpdateCountryProps } from "@/types/types";
+import { Eip3643Context } from "@/contexts/Eip3643Context";
+import { MenuSelect } from "@/components/MenuSelect";
+import {
+  readTokenName,
+  readTokenOwner,
+  writeIdentityRegistryDeleteIdentity,
+  writeIdentityRegistryUpdateCountry,
+  writeIdentityRegistryUpdateIdentity,
+  writeIdentityRegistryAddAgent,
+  watchIdentityRegistryAgentAddedEvent,
+  watchIdentityRegistryAgentRemovedEvent,
+  writeIdentityRegistryRemoveAgent,
+} from "@/services/contracts/wagmiGenActions";
+import { useTokenIdentityRegistry } from "@/hooks/useTokenIdentityRegistry";
+import { useWalletInterface } from "@/services/wallets/useWalletInterface";
+import { WalletInterface } from "@/services/wallets/walletInterface";
+import { WatchContractEventReturnType } from "@/services/contracts/watchContractEvent";
+
+const investorCountriesItems = [
+  {
+    value: 840,
+    label: "US",
+  },
+  {
+    value: 0,
+    label: "Non US",
+  },
+];
+
+export function ManageIdentities() {
+  const [addedAgents, setAddedAgents] = useState([] as Array<string>);
```

Comment:
Lets try to avoid forced type casting with `as` as much as possible

this should suffice
```
useState<string[]>([]);
```